pkg/audit/disqualification_test.go

```diff
@@ -103,3 +108,114 @@ func calcReputation(dossier *overlay.NodeDossier) float64 {
 
 	return alpha / (alpha + beta)
 }
+
+func TestDisqualifiedNodesGetNoDownload(t *testing.T) {
+
+	// - uploads random data
+	// - mark a node as disqualified
+	// - check we don't get it when we require order limit
+
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		satellite := planet.Satellites[0]
+		upl := planet.Uplinks[0]
+
+		err := satellite.Audit.Service.Close()
+		require.NoError(t, err)
+
+		testData := make([]byte, 8*memory.KiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = upl.Upload(ctx, planet.Satellites[0], "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		projects, err := satellite.DB.Console().Projects().GetAll(ctx)
+		require.NoError(t, err)
+		require.Len(t, projects, 1)
+
+		bucketID := []byte(storj.JoinPaths(projects[0].ID.String(), "testbucket"))
+
+		encScheme := upl.GetConfig(satellite).GetEncryptionScheme()
+		cipher := encScheme.Cipher
+		encryptedAfterBucket, err := streams.EncryptAfterBucket(ctx, "testbucket/test/path", cipher, &storj.Key{})
+		require.NoError(t, err)
+
+		lastSegPath := storj.JoinPaths(projects[0].ID.String(), "l", encryptedAfterBucket)
+		pointer, err := satellite.Metainfo.Service.Get(ctx, lastSegPath)
+		require.NoError(t, err)
+
+		disqualifiedNode := pointer.GetRemote().GetRemotePieces()[0].NodeId
+		disqualifyNode(t, ctx, satellite, disqualifiedNode)
+
+		limits, err := satellite.Orders.Service.CreateGetOrderLimits(ctx, upl.Identity.PeerIdentity(), bucketID, pointer)
+		require.NoError(t, err)
+		assert.Len(t, limits, len(pointer.GetRemote().GetRemotePieces())-1)
+
+		for _, orderLimit := range limits {
+			reputable, err := satellite.Overlay.Service.IsVetted(ctx, orderLimit.Limit.StorageNodeId)
+			assert.NoError(t, err)
+			assert.True(t, reputable)
```

Comment:
Why are we checking `IsVetted` for these nodes?

Reply:
Sorry, I forgot to change it